Invalid default generator "Visual Studio 15 2017 Win64 Win64"

Invalid generator is selected with version **5.1.0** on Windows with VS 2017 (double "Win64" appears in the name) by default.

Plain "cmake-js compile" was executed.

**Error output:**

```
c:\burda\projects\wui\tools\com-wui-framework-resource-writer>npm install

resource-writer@0.9.1 install c:\burda\projects\wui\tools\com-wui-framework-resource-writer
cmake-js compile

[ 'C:\\burda\\projects\\wui\\com-wui-framework-builder\\dependencies\\nodejs\\node.exe',
  'c:\\burda\\projects\\wui\\tools\\com-wui-framework-resource-writer\\node_modules\\cmake-js\\bin\\cmake-js',
  'compile' ]
info TOOL Using Visual Studio 15 2017 Win64 Win64 generator.
info CMD CONFIGURE
info RUN cmake "c:\burda\projects\wui\tools\com-wui-framework-resource-writer" --no-warn-unused-cli -G"Visual Studio 15 2017 Win64 Win64" -DCMAKE_JS_VERSION="5.1.0" -DCMAKE_BUILD_TYPE="Release" -DCMAKE_RUNTIME_OUTPUT_DIRECTORY="c:\burda\projects\wui\tools\com-wui-framework-resource-writer\build" -DCMAKE_JS_INC="C:\Users\nxf45876\.cmake-js\node-x64\v10.15.0\include\node" -DNODE_RUNTIME="node" -DNODE_RUNTIMEVERSION="10.15.0" -DNODE_ARCH="x64" -DCMAKE_JS_LIB="C:\Users\nxf45876\.cmake-js\node-x64\v10.15.0\win-x64\node.lib"
Not searching for unused variables given on the command line.
CMake Error: Could not create named generator Visual Studio 15 2017 Win64 Win64

(list of available generators)

```

--- a/lib/es6/toolset.js
+++ b/lib/es6/toolset.js
@@ -177,14 +177,23 @@ Toolset.prototype._getTopSupportedVisualStudioGenerator = async(function*() {
     let result = null;
     for (let gen of list) {
         let found = /^visual studio (\d+)/i.exec(gen);
-        if (found) {
-            let ver = parseInt(found[1]);
-            if (ver > maxVer) {
-                if (yield vsDetect.isInstalled(ver + ".0")) {
-                    result = this.targetOptions.isX64 ? (gen + " Win64") : gen;
-                    maxVer = ver;
-                }
-            }
+        if (!found) {
+            continue;
+        }
+
+        let ver = parseInt(found[1]);
+        if (ver <= maxVer) {
+            continue;
+        }
+
+        const is64Bit = gen.endsWith("Win64");
+        if ((this.targetOptions.isX86 && is64Bit) || (this.targetOptions.isX64 && !is64Bit)) {
+            continue;
+        }
+
+        if (yield vsDetect.isInstalled(ver + ".0")) {
+            result = gen;
+            maxVer = ver;
         }
     }
     return result;

